src/Text/Gigaparsec/Char.hs

```diff
@@ -94,7 +94,16 @@ satisfy :: (Char -> Bool) -- ^ the predicate, @pred@, to test the next character
                           -- exist.
         -> Parsec Char    -- ^ a parser that tries to read a single character @c@, such that @pred c@
                           -- is true, or fails.
-satisfy _ = Internal.Parsec undefined --TODO:
+satisfy test = Internal.Parsec $ \st ok err ->
+  case Internal.input st of
+    (x: xs) | test x  ->
+      ok x ((updatePos st x) { Internal.input = xs, Internal.consumed = True })
```

Comment:
Shame about the double allocation, but oh well...

Reply:
does GHC not inline/concatenate this?

Reply:
_maybe_? You'd have to look at the core to know for sure... I wonder if marking `updatePos` as `{-# INLINE updatePos #-}` would help it along the way?

Reply:
could always handwrite it or ... template haskell it

Reply:
could handwrite it if you wanted to, TH is overkill :)